src/lib/chips/chip-list.ts

```diff
@@ -109,59 +147,68 @@ export class MdChipList implements AfterContentInit, OnDestroy {
     this._selectable = coerceBooleanProperty(value);
   }
 
+  /** Associates an HTML input element with this chip list. */
+  registerInput(inputElement: HTMLInputElement) {
+    this._inputElement = inputElement;
+  }
+
   /**
-   * Programmatically focus the chip list. This in turn focuses the first
-   * non-disabled chip in this chip list.
+   * Focuses the the first non-disabled chip in this chip list, or the associated input when there
+   * are no eligible chips.
    */
   focus() {
-    // TODO: ARIA says this should focus the first `selected` chip.
-    this._keyManager.setFirstItemActive();
+    // TODO: ARIA says this should focus the first `selected` chip if any are selected.
+    if (this.chips.length > 0) {
+      this._keyManager.setFirstItemActive();
+    } else {
+      this._focusInput();
+    }
   }
 
-  /** Passes relevant key presses to our key manager. */
-  _keydown(event: KeyboardEvent) {
-    let target = event.target as HTMLElement;
-
-    // If they are on a chip, check for space/left/right, otherwise pass to our key manager
-    if (target && target.classList.contains('mat-chip')) {
-      switch (event.keyCode) {
-        case SPACE:
-          // If we are selectable, toggle the focused chip
-          if (this.selectable) {
-            this._toggleSelectOnFocusedChip();
-          }
-
-          // Always prevent space from scrolling the page since the list has focus
-          event.preventDefault();
-          break;
-        case LEFT_ARROW:
-          this._keyManager.setPreviousItemActive();
-          event.preventDefault();
-          break;
-        case RIGHT_ARROW:
-          this._keyManager.setNextItemActive();
-          event.preventDefault();
-          break;
-        default:
-          this._keyManager.onKeydown(event);
-      }
+  /** Attempt to focus an input if we have one. */
+  _focusInput() {
+    if (this._inputElement) {
+      this._inputElement.focus();
     }
   }
 
-  /** Toggles the selected state of the currently focused chip. */
-  protected _toggleSelectOnFocusedChip(): void {
-    // Allow disabling of chip selection
-    if (!this.selectable) {
+  /**
+   * Pass events to the keyboard manager. Available here for tests.
+   */
+  _keydown(event: KeyboardEvent) {
+    let code = event.keyCode;
+    let target = event.target as HTMLElement;
+    let isInputEmpty = _isInputEmpty(target);
+    let isRtl = this._dir && this._dir.value == 'rtl';
+
+    let isPrevKey = (code == (isRtl ? RIGHT_ARROW : LEFT_ARROW));
+    let isNextKey = (code == (isRtl ? LEFT_ARROW : RIGHT_ARROW));
+    let isBackKey = (code == BACKSPACE || code == DELETE || code == UP_ARROW || isPrevKey);
+    let isForwardKey = (code == DOWN_ARROW || isNextKey);
+    // If they are on an empty input and hit backspace/delete/left arrow, focus the last chip
+    if (isInputEmpty && isBackKey) {
```

Comment:
Just curious: how did you decide to disable moving "left" into the chips when the input isn't empty? material1 and gmail do this same thing, but Drive lets you arrow back through the chips even when there is still text.